### DISCUSSION TITLE
2024 Cohort 2 - Group 11 Discussion

### DISCUSSION THREAD

**Discussion by sharon-digipres** [Digital-Preservation-Coalition/PythonStudyGroups#37] [General]
This is the discussion thread for those in **Group 11** of the July-December 2024 Cohort

**Comment by harrylloyd-bl** [comment]
Hi Group 11 if you need anything @ me or use the [questions](https://github.com/Digital-Preservation-Coalition/PythonStudyGroups/discussions/categories/questions) section 😄

**Comment by MalcolmMathieson** [comment]
Hi Group 11! Thanks for 2 great meetings so far. Just to recap, I'm Malcolm, Archivist at Amnesty International. I'm hoping to use Python to automate tasks like folder creation, folder renaming and batch processing PDFs

**Comment by LukasHech** [comment]
Hi Group 11! Lukas here from the IAEA Archives. Thank you for your efforts in organizing the meetings and setting up our communication channels & apologies for not having been very responsive, I've been quite occupied at work. Is anyone else coming to iPRES next week? I'll be there, onsite (and thus not in the next group session). Cheers!

**Reply by miten-mistry** [reply]
I am not coming to iPRES but it did look an interesting conference. I was at the ARA conference last month which was a lot of fun.

**Comment by miten-mistry** [comment]
Hi group 11. just doing a post on the discussion, to see what it is like. Thanks again to everyone for setting it up everything. 
I have been doing some of the intro stuff and exploring some online learning resources that I can mention in our next meeting. looking forward to seeing anyone that can make it tomorrow.
`testing what the code formatting looking like`

**Comment by jennyhuffman42** [comment]
Hey there everyone! I'm Jenny from Tennessee Tech Archives. Another month already? Let's try this code out:
print ("hello" + " world")

**Reply by jennyhuffman42** [reply]
well, apparently all of the dots are not clicking for me. Alas, see you all tomorrow!

**Comment by gruinardhandshake** [comment]
Hiya! This is Jennifer, University of Rhode Island. I haven't tested any code yet, but things are starting to make a little more sense ʕ•ᴥ•ʔ

**Comment by GeoffBurnsy** [comment]
Is this the group I joined? I can't remember.
Did we choose a group objective? I want to try and do more but don't really have a specific goal in mind at the moment.

**Reply by gruinardhandshake** [reply]
Hi Geoff! Sorry for the late response. We've met a few times and our next meeting is Nov. 20th if you want to join! Our objectives are more individually based (catching-up with testing code, trying out AI to write code, etc.).

**Comment by MalcolmMathieson** [comment]
@harrylloyd-bl Hi Harry, I'm getting a the following error trying to install the Pillow library using the terminal on VS code. Any help would be great.

PS C:\Digitisation Scripting> pip install pillow
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get 
local issuer certificate (_ssl.c:1006)'))': /simple/pillow/

The error repeats 5 times then I get the below message

Could not fetch URL https://pypi.org/simple/pillow/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pillow/ (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1006)'))) - skipping
ERROR: Could not find a version that satisfies the requirement pillow (from versions: none)
ERROR: No matching distribution found for pillow
Could not fetch URL https://pypi.org/simple/pip/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pip/ (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate 
(_ssl.c:1006)'))) - skipping
PS C:\Digitisation Scripting>

**Reply by GeoffBurnsy** [reply]
I'm not sure if this is the same problem I had. I'm using work PCs without Admin rights so I had to manually download the .whl files for the module from the website and install them the following way.

In PowerShell, cd to the folder with the .whl files in then type something like:
`py -m pip install pillow.whl --no-index --trusted-host=None --find-links="C:\the folder with the files in"`

It didn't work at first because some extra supporting .whl files needed to be added.

**Reply by harrylloyd-bl** [reply]
Malcolm sounds like it might be an older version of pip having problems with your organisation's firewall. What version of pip are you running  `pip --version`?   
If it's less than ~23 try updating it `python -m pip install --trusted-host pypi.org --trusted-host pypi.python.org --trusted-host files.pythonhosted.org --upgrade pip setuptools`

A fairly common part of debugging best practice is to check your package versions and update if needed.

**Reply by MalcolmMathieson** [reply]
@harrylloyd-bl thanks for the tip. My pip version is 23.2.1

**Reply by MalcolmMathieson** [reply]
@GeoffBurnsy Thanks Geoff. There seem to be many different .whl files to choose from. How do I decide which ones I need?

![image](https://github.com/user-attachments/assets/933d1ff9-07eb-4b06-8cc5-447b2ea50634)

**Reply by harrylloyd-bl** [reply]
You'll need to pick the one for the correct python version (e.g. the pp310-pypy310 is for Python 3.10), for your operating system (the win/macosx/manylinux part) and, if necessary, for your architecture (e.g. arm64/x86_64 for mac)

If your pip is in date then it's probably an issue with the company firewall. You could
- try chatting to IT to see if they can unblock pypi
- mark pypi as trusted when you call pip install as they do [here](https://stackoverflow.com/questions/64311305/ssl-certificate-verify-failed-error-while-using-pip)
- use the wheels as above, which is defs a valid if faffy way to do it!

**Reply by MalcolmMathieson** [reply]
Thanks Harry. The second bullet point on making pypi trusted worked. I am now encountering separate errors but have at least moved forward a little bit!
